conftest.py

```diff
@@ -15,3 +20,29 @@ def pytest_collection_modifyitems(config, items):
     for item in items:
         if "latest_runtime" in item.keywords:
             item.add_marker(skip_latest)
+
+
+@pytest.fixture
+def software():
+    return os.environ.get(
+        "COILED_SOFTWARE_NAME",
+        f"dask-engineering/coiled_dist-py{sys.version_info[0]}{sys.version_info[1]}",
+    )
+
+
+@pytest.fixture
+def small_cluster(software):
```

Comment:
Do we want to allow multiple tests to use the same cluster? Should be be restarted each time? Recreated each time? Wondering about fixture scope.

Reply:
Maybe.  This is what they're doing with the H2O benchmarks.  We'll want things clean, but maybe we can achieve that with restart, scale, and wait_for_workers.

The platform integration tests do a single cluster per test, and they just run all of the tests in parallel.  It's less efficient, but works ok.

I'm open here.  I think that we'll probably want to iterate.

Reply:
I'm suggesting we start with module-scoped cluster fixtures + function-scoped client fixtures with restarts over in https://github.com/coiled/coiled-runtime/pull/40. This is something that seems to work for our current needs and, as @mrocklin said, we can iterate as needed. 

+1 for parallelizing tests. Opening a separate issue for that.

Reply:
In platform integration tests we asked pytest-xdist to parallelize between modules, but not within them.  That would work will with the module-scoped cluster fixtures.